xrpl/utils/tx_parser/utils.py

```diff
@@ -0,0 +1,850 @@
+
+    pass
+
+
+def is_valid_metadata(metadata: Dict[str, Any]) -> None:
+    """Check if the given metadata is valid.
+
+    Args:
+        metadata (Dict[str, Any]): The transactions metadata
+
+    Raises:
+        XRPLMetadataException: If field Account is missing.
+        XRPLMetadataException: If field Account is missing.
+        XRPLMetadataException: If field meta is missing.
+        XRPLMetadataException: If field AffectedNodes is missing or is empty.
+    """
+    if "transaction" in metadata:
+        if "Account" not in metadata["transaction"]:
+            raise XRPLMetadataException(
+                "Metadata incomplete: Metadata field 'Account' must be included."
+            )
+    elif "Account" not in metadata:
+        raise XRPLMetadataException(
+            "Metadata incomplete: Metadata field 'Account' must be included."
+        )
+
+    if "meta" not in metadata:
+        raise XRPLMetadataException(
+            "Metadata incomplete: Metadata field 'meta' must be included."
+        )
+
+    if "AffectedNodes" not in metadata["meta"] or not metadata["meta"]["AffectedNodes"]:
+        raise XRPLMetadataException("Metadata incomplete: No nodes provided.")
+
+
+class NormalizedNode:
+    """A standard format for nodes."""
+
+    def __init__(
+        self: Any,
+        diffType: str,
+        entryType: str,
+        ledgerIndex: str,
+        newFields: Any,
+        finalFields: Any,
+        previousFields: Any,
+    ) -> None:
+        """
+        Args:
+            diffType (str): Node type (ModifiedNode, CreatedNode or DeletedNode).
+            entryType (str): Entry type (e.g. Offer, AccountRoot, …).
+            ledgerIndex (str): Ledger index.
+            newFields (Any): New fields.
+            finalFields (Any):
+            Fields after the transaction occurred.
+            previousFields (Any):
+            Fields before the transaction occurred.
+        """
+        self.diff_type = diffType
+        self.entry_type = entryType
+        self.ledger_index = ledgerIndex
+        self.new_fields = newFields
+        self.final_fields = finalFields
+        self.previous_fields = previousFields
+
+
+class NewFields:
+    """NewFields"""
+
+    pass
+
+
+class FinalFields:
+    """FinalFields"""
+
+    pass
+
+
+class PreviousFields:
+    """PreviousFields"""
+
+    pass
+
+
+def field_factory(
+    field_state: Union[
+        Literal["NewFields"], Literal["FinalFields"], Literal["PreviousFields"]
+    ],
+    fields: Any,
+) -> Union[NewFields, FinalFields, PreviousFields]:
+    """Set the attributes for 'NewFields', 'FinalFields' and 'ModifiedFields'.
+
+    Args:
+        field_state:
+            State of the field.
+        fields:
+            A dictionary of all fields that field state has.
+            E.g.: {'Balance': '62537659', 'OwnerCount': 16, 'Sequence': 67702065}
+
+    Returns:
+        The full field state with all its fields as objects.
+    """
+
+    def factory(name: str) -> Any:
```

Comment:
Would creating a base class you return from factory help cut down on the use of Any? 

(The main reason being that Mypy loses a lot of effectiveness the more `Any`s enter the code because they multiply)